### PR TITLE
move test filtering for aptos-framework

### PR DESCRIPTION
**Pull request by zjma**
A way to filter move unit tests and make aptos-framework development slightly easier.

### CODE REVIEW

**Comment by zjma** [comment]
> That's amazing. Is there a way to do this via a `cargo test` flag instead of an environment variable?

envvar is the best I can come up with so far...

File: aptos-move/framework/README.md [hunk 1]
```diff
@@ -47,3 +47,14 @@ The overall structure of the Aptos Framework is as follows:
...
 ```
+
+## Move unit test selection
+In this package, unit test run triggered by `cargo test <filter>` will only select rust test wrappers
+(for example, those marked as `#[test]` in `tests/move_unit_test.rs`).
+
+To select the actual test cases in Move, use the following command.
+```bash
+MOVE_TEST_FILTER=<move-test-filter> cargo test <the-rest-params>
+```
```
Comment (gregnazario) [inline aptos-move/framework/README.md:58]: We should stay away from using environment variables to control configuration, and instead add a flag `--filter` to filter tests

Comment (movekevin) [reply aptos-move/framework/README.md:58]: +1. I appreciate the attempt here but this is what aptos move test -f filter is designed for

Comment (movekevin) [reply aptos-move/framework/README.md:58]: I see. Can’t you do cargo run -p aptos — move test …?

Comment (alinush) [reply aptos-move/framework/README.md:58]: That will take a really long time to recompile the CLI...

Comment (davidiw) [reply aptos-move/framework/README.md:58]: use path/to/aptos-core/target/debug/aptos move test

Comment (alinush) [reply aptos-move/framework/README.md:58]: How does that address the long time to recompile after modifying a Rust native?

Comment (davidiw) [reply aptos-move/framework/README.md:58]: Ahhh... fwiw, you can filter by the framework package.

I do like this direction, just not entirely sure how easy it is to get the flags from test and route them appropriately :grimacing:

**Pull request closed without merging**